CharacterSheet.API/DataAccess/D20Repo.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using ClassLibrary;
+using Microsoft.EntityFrameworkCore;
+
+namespace DataAccess
+{
+    class D20Repo : IRepo
+    {
+
+        private readonly D20CharacterDatabaseContext _db;
+
+        /// <summary>
+        /// Initializes a new Character repository given a suitable Entity Framework DbContext.
+        /// </summary>
+        /// <param name="db">The DbContext</param>
+        /// 
+        public D20Repo(D20CharacterDatabaseContext db)
+        {
+            _db = db ?? throw new ArgumentNullException(nameof(db));
+        }
+
+        public void AddGM(int CampID, int UserID)
+        {
+            _db.Gmjunction.Add(new Gmjunction()
+            {
+                CampaignId = CampID,
+                Gmid = UserID
+            });
+        }
+
+        public Campaign CampDetails(int CampID)
+        {
+            return _db.Campaign.Find(CampID);
+        }
+
+        public IEnumerable<Campaign> CampList()
+        {
+            return _db.Campaign.Include(c => c.Characters).Include(c => c.Gmjunction);
+        }
+
+        public IEnumerable<Character> CharacterList()
+        {
+            List<Character> ret = (List<Character>) _db.Characters.Include(c => c.Classes).Include(c => c.Feats).Include(c => c.Inventory).Include(c => c.Skills).Include(c => c.SpellJunction).Include(c => c.SpellSlots).GetEnumerator();
```

Comment:
It wasn't giving me the option, possibly because I had not included Linq yet. I'll go back on the next runthrough and see if it allows me to do .ToListAsync().